analyze-ms-stats/calc-manuscript-stats.py

```diff
@@ -0,0 +1,126 @@
+import json
+import pandas as pd
+import matplotlib
+import argparse
+import time
+import os.path
+import multiprocessing
+import subprocess
+
+def analyze_commit(commit):
+    """Access files and data in variables.json associated with each commit
+    Accepts commit ID as string
+    Returns list of 5 statistics"""
+
+    variablesCommand = "git show " + commit + ":./variables.json"
+    variables = json.loads(subprocess.getoutput(variablesCommand))
+
+    date = variables['pandoc']['date-meta']
+    clean_date = variables['manubot']['date']
+    num_authors = len(variables['manubot']['authors'])
+    word_count = variables['manubot']['manuscript_stats']['word_count']
+
+    # Access files and data in references.json associated with each commit
+    referencesCommand = "git show " + commit + ":./references.json"
+    references = json.loads(subprocess.getoutput(referencesCommand))
+    num_ref = len(references)
+
+    return([date, clean_date, num_authors, word_count, num_ref])
+
+def main(args):
+    '''Extract statistics from the output branch log'''
+
+    start_time = time.time()
+    print(f"Using {multiprocessing.cpu_count()} CPUs")
+
+    # Read in list of all commits on this branch
+    with open(args.commit_list, "r") as commitFile:
+        commits = [c.strip() for c in commitFile.read().splitlines()]
+
+    # If this analysis has been run before, load in the list of commits analyzed
+    # and only analyze new commits
+    # Assumes no commits will be added retrospectively (to take advantage of linearity)
```

Comment:
One way to remove this assumption could be to create a `set` of old and new commits (I understand these are commits' hashes). Then you can process the commits that are new only. Assumbing both `commits` and `oldCommits` are both lists of commits' hashes, it would be something like: `commits_to_be_processed = set(commits) - set(oldCommits)`.